### PR TITLE
Fix Issue #48 - Endless upload loop when no network

### PR DESCRIPTION
**Pull request by justindarc**

### CODE REVIEW

File: Telemetry/TelemetryScheduler.swift [hunk 1]
```diff
@@ -45,20 +45,29 @@ public class TelemetryScheduler {
...
-                    self.incrementDailyUploadCountForPingType(pingType)
-                    uploadNextPing()
                 } else {
-                    // Don't delete this ping even though we couldn't upload it right now. Just continue on
-                    // to the next ping.
-                    uploadNextPing()
+                    if error == nil {
+                        // Move the ping to the "retry" directory since we encountered a temporary server
+                        // issue.
+                        pingSequence.moveToRetryDirectory()
```
Comment (garvankeeley) [inline Telemetry/TelemetryScheduler.swift]: why not bail out here?

File: Telemetry/TelemetryStorage.swift [hunk 2]
```diff
@@ -11,37 +11,51 @@ import Foundation
...
+        self.files = files
     }
 
     func next() -> TelemetryPing? {
-        guard let directoryEnumerator = self.directoryEnumerator else {
-            return nil
-        }
+        index += 1
+
+        while files.count > index {
```
Comment (garvankeeley) [inline Telemetry/TelemetryStorage.swift]: why not a for loop so the index is incremented always?

File: Telemetry/TelemetryStorage.swift [hunk 3]
```diff
@@ -11,37 +11,51 @@ import Foundation
...
+            let url = files[index]
+            
+            var isDirectory: ObjCBool = false
+
+            guard FileManager.default.fileExists(atPath: url.path, isDirectory: &isDirectory), !isDirectory.boolValue else {
+                index += 1
+                continue
+            }
+
+            currentPingFile = url
```
Comment (garvankeeley) [inline Telemetry/TelemetryStorage.swift]: is this var needed now that there is a current index? 
Could it be a getter to return files[index]

**Comment by justindarc** [comment]
Closing this in favor of PR #50

**Pull request closed without merging**